File: app/src/main/java/com/letianpai/robot/desktop/ui/view/DeleteConfirmView.java
package com.letianpai.robot.desktop.ui.view;

import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageInstaller;
import android.text.TextUtils;
import android.util.AttributeSet;
import android.view.View;
import android.widget.ImageView;
import android.widget.LinearLayout;

import androidx.annotation.Nullable;

import com.letianpai.robot.desktop.R;
import com.letianpai.robot.desktop.callback.AppMenuUpdateCallback;
import com.letianpai.robot.desktop.manager.RobotAppListManager;
import com.letianpai.robot.desktop.ui.activity.UserAppMainActivity;

public class DeleteConfirmView extends LinearLayout {
    private Context mContext;
    private ImageView ivCancel;
    private ImageView ivConfirm;
    private BackButton ivBack;

    public DeleteConfirmView(Context context) {
        super(context);
    }

    public DeleteConfirmView(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        initView(context);
    }

    public DeleteConfirmView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        initView(context);
    }

    public DeleteConfirmView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        initView(context);
    }


    private void initView(Context context) {
        this.mContext = context;
        inflate(mContext, R.layout.app_confirm_view,this);
        initViews();
        addListeners();
    }

    private void initViews() {
        ivConfirm = findViewById(R.id.iv_confirm);
        ivCancel = findViewById(R.id.iv_cancel);
        ivBack = findViewById(R.id.back_firm_view);
    }

    private void addListeners() {
        ivConfirm.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                responseConfirm();
            }
        });
        ivCancel.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                responseCancel();
            }
        });
        ivBack.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                responseBack();
            }
        });

    }

    private void responseBack() {
        responseCancel();
    }

    private void responseCancel() {
        ((UserAppMainActivity)mContext).showUserView();
    }

    protected void responseConfirm(){
        if (((UserAppMainActivity) mContext).getAppMenuInfo() != null && !TextUtils.isEmpty(((UserAppMainActivity) mContext).getAppMenuInfo().getPackageName())) {
            String packageName = ((UserAppMainActivity) mContext).getAppMenuInfo().getPackageName();
            unInstallApp(packageName);
//            RobotAppListManager.getInstance(mContext).updateAppMenuList();
            AppMenuUpdateCallback.getInstance().setAppStatusUpdate(packageName,1);
            ((UserAppMainActivity)mContext).closeActivity1();
        }
    };

    private void unInstallApp(String packageName) {
        Intent intent = new Intent();
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        PendingIntent sender = PendingIntent.getActivity(mContext, 0, intent, 0);
        PackageInstaller mPackageInstaller = mContext.getPackageManager().getPackageInstaller();
        mPackageInstaller.uninstall(packageName, sender.getIntentSender());// 卸载APK
    }
}

File: app/src/main/java/com/letianpai/robot/desktop/ui/view/DeleteConfirmView.kt
package com.letianpai.robot.desktop.ui.view

import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.text.TextUtils
import android.util.AttributeSet
import android.widget.ImageView
import android.widget.LinearLayout
import com.letianpai.robot.desktop.R
import com.letianpai.robot.desktop.callback.AppMenuUpdateCallback
import com.letianpai.robot.desktop.ui.activity.UserAppMainActivity

class DeleteConfirmView : LinearLayout {
    private var mContext: Context? = null
    private var ivCancel: ImageView? = null
    private var ivConfirm: ImageView? = null
    private var ivBack: BackButton? = null

    constructor(context: Context?) : super(context)

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        initView(context)
    }

    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
        initView(context)
    }

    constructor(
        context: Context,
        attrs: AttributeSet?,
        defStyleAttr: Int,
        defStyleRes: Int
    ) : super(context, attrs, defStyleAttr, defStyleRes) {
        initView(context)
    }


    private fun initView(context: Context) {
        this.mContext = context
        inflate(mContext, R.layout.app_confirm_view, this)
        initViews()
        addListeners()
    }

    private fun initViews() {
        ivConfirm = findViewById(R.id.iv_confirm)
        ivCancel = findViewById(R.id.iv_cancel)
        ivBack = findViewById(R.id.back_firm_view)
    }

    private fun addListeners() {
        ivConfirm!!.setOnClickListener { responseConfirm() }
        ivCancel!!.setOnClickListener { responseCancel() }
        ivBack!!.setOnClickListener { responseBack() }
    }

    private fun responseBack() {
        responseCancel()
    }

    private fun responseCancel() {
        (mContext as UserAppMainActivity).showUserView()
    }

    protected fun responseConfirm() {
        if ((mContext as UserAppMainActivity).appMenuInfo != null && !TextUtils.isEmpty((mContext as UserAppMainActivity).appMenuInfo!!.packageName)) {
            val packageName = (mContext as UserAppMainActivity).appMenuInfo!!.packageName
            unInstallApp(packageName!!)
            //            RobotAppListManager.getInstance(mContext).updateAppMenuList();
            AppMenuUpdateCallback.instance.setAppStatusUpdate(packageName, 1)
            (mContext as UserAppMainActivity).closeActivity1()
        }
    }

    private fun unInstallApp(packageName: String) {
        val intent = Intent()
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
        val sender = PendingIntent.getActivity(mContext, 0, intent, PendingIntent.FLAG_IMMUTABLE)
        val mPackageInstaller = mContext!!.packageManager.packageInstaller
        mPackageInstaller.uninstall(packageName, sender.intentSender) // 卸载APK
    }
}
